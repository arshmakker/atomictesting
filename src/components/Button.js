import React from 'react'

function Button({
    classButton = "",
    title = "Press Me",
    onButtonPress = () => { console.log('Button Pressed') }
}) {
    return (
        <button className={classButton} onClick={onButtonPress}>{title}</button>
    )
}

export default Button;