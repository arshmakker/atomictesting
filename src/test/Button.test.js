import React from 'react'
import Button from '../components/Button'
import renderer from 'react-test-renderer'

const classButton = "sampleClass"
const title = "Sample Title"


it("Renders the Button correctly with no props ", () => {
    const tree = renderer
        .create(<Button/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})
it("Renders the Button correctly with test props", () => {

    const tree = renderer
        .create(<Button classButton={classButton}
            title={title} ></Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})