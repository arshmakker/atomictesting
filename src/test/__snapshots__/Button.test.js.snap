// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Button correctly with no props  1`] = `
<button
  className=""
  onClick={null}
>
  Press Me
</button>
`;

exports[`Renders the Button correctly with test props 1`] = `
<button
  className="sampleClass"
  onClick={null}
>
  Sample Title
</button>
`;
